rueidisaside/aside.go

```diff
@@ -197,6 +204,7 @@ func (c *Client) Close() {
 const PlaceholderPrefix = "rueidisid:"
 
 var (
-	delkey = rueidis.NewLuaScript(`if redis.call("GET",KEYS[1]) == ARGV[1] then return redis.call("DEL",KEYS[1]) else return 0 end`)
-	setkey = rueidis.NewLuaScript(`if redis.call("GET",KEYS[1]) == ARGV[1] then return redis.call("SET",KEYS[1],ARGV[2],"PX",ARGV[3]) else return 0 end`)
+	delkey    = rueidis.NewLuaScript(`if redis.call("GET",KEYS[1]) == ARGV[1] then return redis.call("DEL",KEYS[1]) else return 0 end`)
+	setkey    = rueidis.NewLuaScript(`if redis.call("GET",KEYS[1]) == ARGV[1] then return redis.call("SET",KEYS[1],ARGV[2],"PX",ARGV[3]) else return 0 end`)
+	setKeyLua = rueidis.NewLuaScript(`if redis.call("SET", KEYS[1], ARGV[1], "NX", "PX", ARGV[2]) then return KEYS[1] else return nil end`)
```

Comment:
We should keep the script the same function as the SET NX GET. It should return nil when set and return the existing value when not set.

Reply:
Hi @LegendHawk27, are you still interested in working on this?

Reply:
Need some help to solve this issue

Reply:
What help do you need?